backend/api/Services/TimeseriesServiceSqlLite.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Linq.Expressions;
+using Api.Controllers.Models;
+using Api.Database.Context;
+using Api.Database.Models;
+using Api.Utilities;
+
+namespace Api.Services
+{
+    /// <summary>
+    /// Uses only dotnet ef core, instead of the Npgsql package needed for the PostgreSql database
+    /// Cannot insert because it is a keyless entity
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1309:Use ordinal StringComparison",
+        Justification = "EF Core refrains from translating string comparison overloads to SQL"
+    )]
+    public class TimeseriesServiceSqlLite : ITimeseriesService
+    {
+        private readonly FlotillaDbContext _context;
+
+        public TimeseriesServiceSqlLite(FlotillaDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IEnumerable<T>> ReadAll<T>(
+            TimeseriesQueryStringParameters queryStringParameters
+        ) where T : TimeseriesBase
+        {
+            var query = _context.Set<T>().AsQueryable();
+            var filter = ConstructFilter<T>(queryStringParameters);
+
+            query = query.Where(filter);
+            query = query.OrderByDescending(timeseries => timeseries.Time);
+
+            return await PagedList<T>.ToPagedListAsync(
+                query.OrderByDescending(timeseries => timeseries.Time),
+                queryStringParameters.PageNumber,
+                queryStringParameters.PageSize
+            );
+        }
+
+        // Cannot use Entity framework to insert keyless entities into the timeseries database
+        public async Task<T> Create<T>(T newTimeseries) where T : TimeseriesBase
+        {
+            return newTimeseries;
```

Comment:
Do we want to await something in this function as it returns a `Task`

Reply:
There is nothing to await, but it needs to return a task because of the class it overrrides